tests/integration-tests/clusters_factory.py

```diff
@@ -113,6 +128,7 @@ def delete(self, delete_logs=False):
                 error = "Cluster deletion failed for {0} with output: {1}".format(self.name, result.stdout)
                 logging.error(error)
                 raise Exception(error)
+            self._delete_resource_by_stack_id_tag()
```

Comment:
shall we move this before the if so that we try this deletion also in case of a delete failred?

Reply:
Done.
My concern was it would be possible to delete root volume attached to an instance if the stack deletion failed. I tried to call boto3 `delete_volume` to delete an attached root volume and it just threw an exception without deleting the volume. So this new change is safe.